src/gluonts/mx/trainer/_base.py

```diff
@@ -404,6 +408,8 @@ def loop(  # todo call run epoch
                         logger.info(
                             f"Number of parameters in {net_name}: {num_model_param}"
                         )
+                    if not should_continue:
+                        break
```

Comment:
This doesn't set the outer `should_continue` and thus we will call `loop` again and again.

Reply:
I fixed this by setting ```self.halt=True``` before we break from the loop.